pkg/controller/master/upgrade/version_syncer.go

```diff
@@ -149,17 +149,21 @@ func (s *versionSyncer) getExtraInfo() (map[string]string, error) {
 }
 
 func (s *versionSyncer) syncVersions(resp CheckUpgradeResponse, currentVersion string) error {
-	if err := s.cleanupVersions(currentVersion); err != nil {
+	if err := s.cleanupVersions(currentVersion, resp.Versions); err != nil {
 		return err
 	}
```

Comment:
it is not related to this PR,

but the `syncVersions` is not stopped when upgrade is ongoing; we had other PR to do not show the newly available versions after Harvester is just upgraded (with new currentVersion) but the upgrade is not finished

it looks the better solution is to stop sync & `cleanupVersions` to avoid potential effect against upgrade

Reply:
the upgrade controller does not depend on the version object to exist. The only time version object is referred is when the upgrade controller needs to create a repo image. If an upgrade has create the image then deleting the version will have no real impact. If version is deleted before that then upgrade will fail which may not be a bad idea since the version would have been recalled for a specific reason